crates/orchestrator/src/jobs/da_job/mod.rs

```diff
@@ -570,6 +590,71 @@ pub mod test {
         assert_eq!(data, deserialize_data);
     }
 
+    #[rstest]
+    #[case(vec![], vec![], vec![], 0)] // Empty case
+    #[case(
+        vec![(Felt::from(1), Felt::from(10)), (Felt::from(2), Felt::from(20))],
+        vec![],
+        vec![],
+        2
+    )] // Only nonces
+    #[case(
+        vec![],
+        vec![],
+        vec![(Felt::from(1), vec![1]), (Felt::from(2), vec![2])],
+        2
+    )] // Only deployed
+    #[case(
+        vec![(Felt::from(1), Felt::from(10))],
+        vec![(Felt::from(1), vec![(Felt::from(1), Felt::from(100))])],
+        vec![(Felt::from(1), vec![1])],
+        1
+    )] // Overlapping addresses
+    #[case(
+        vec![(Felt::from(1), Felt::from(10)), (Felt::from(1), Felt::from(20))],
+        vec![],
+        vec![(Felt::from(1), vec![1]), (Felt::from(1), vec![2])],
+        1
+    )] // Duplicate addresses
+    fn test_refactor_state_update(
+        #[case] nonces: Vec<(Felt, Felt)>,
+        #[case] storage_diffs: Vec<(Felt, Vec<(Felt, Felt)>)>,
+        #[case] deployed_contracts: Vec<(Felt, Vec<u8>)>,
+        #[case] expected_storage_count: usize,
+    ) {
+        // Create initial state
+        let mut state_diff = create_state_diff(nonces, storage_diffs.clone(), deployed_contracts);
+        let initial_storage = state_diff.storage_diffs.clone();
+
+        // Run the function
+        refactor_state_update(&mut state_diff);
+
+        // Verify results
+        assert!(verify_addresses_have_storage_diffs(&state_diff));
+
+        // Verify no duplicate addresses in storage_diffs
+        let unique_addresses: HashSet<_> = state_diff.storage_diffs.iter().map(|item| &item.address).collect();
```

Comment:
this is being calculated twice right? once inside `verify_addresses_have_storage_diffs` and once here?